{% extends 'base.html' %}

{% block title %}
    CoNekT@LIS : Co-expression Network Toolkit at LIS
{% endblock %}

{% block container %}
<div class="top-pad">
    <ol class="breadcrumb">
        <li class="active">Home</li>
    </ol>

    <div class="row">
        <div class="col-lg-12">

            <div style="overflow:hidden">
                <!--<img src="{{ url_for('static', filename='img/icon-192x192.png') }}" style="float:left;margin-right:25px;margin-top:-20px;margin-bottom:20px" />-->
                <a href="https://legumeinfo.org"><img src="{{ url_for('static', filename='img/legumeinfo_logo.png') }}" style="float:left;margin-right:25px;margin-top:-20px;margin-bottom:20px" /></a>
                <div><!--<h1>Welcome to CoNekT<br /> <small>a platform dedicated to the visualization and analysis of plant co-expression and co-function networks.</small></h1>-->
			
<!-- ========================================================= -->
<h2><small>A toolkit to visualize co-expression of genes across species in selected datasets</small></h2>			
The current version was built with a few species using the following annotations-- <b>arahy.Tifrunner.gnm2.ann1, cicar.CDCFrontier.gnm1.ann1, glyma.Wm82.gnm2.ann1, phavu.G19833.gnm1.ann1, vigun.IT97K-499-35.gnm1.ann1</b> and accepts only those gene IDs. See <a href="#datasets">below</a> for the source of data. 

<style>p {  text-indent: 30px; }</style>
<h3>Commonly Used Tools</h3>

<b style="font-size:larger"><!--<a href="/conekt/search/advanced"> -->Expression Profile of a Single Gene and Profile Neighbors:</b>
<p>Paste an exact gene ID from the appropriate annotation to display its expression profile and below it also <b>find similarly expressed genes(profile neighbors)</b> in section 'Co-expression Networks' in the result page. </p>
<form id="gotogeneform" oninput="urlgene = parturl.value +''+ geneid.value.trim(); gotogeneform.action= urlgene; genelink.href= urlgene; genelink.innerHTML = urlgene;" >
    <input id="parturl" type="hidden" value="/conekt/sequence/find_forLIS/" />
    <b>geneID (Full exact ID):</b> <input type="text" id="geneid" defaultValue="arahy.Tifrunner.gnm2.ann1.0003EL"  size="50" /> 
    <input type="submit"  value="submit"/><br/>
    <b>Examples:</b> &nbsp;&nbsp;&nbsp; arahy.Tifrunner.gnm2.ann1.0003EL &nbsp;&nbsp;&nbsp;  cicar.CDCFrontier.gnm1.ann1.Ca_06316 &nbsp;&nbsp;&nbsp;&nbsp; glyma.Wm82.gnm2.ann1.Glyma.06G321800 &nbsp;&nbsp;&nbsp; phavu.G19833.gnm1.ann1.Phvul.001G000500 &nbsp;&nbsp;&nbsp; vigun.IT97K-499-35.gnm1.ann1.Vigun06g115800  
</form>
<font style="color:gray">URL for the gene:</font> <a id="genelink" ></a> <br/>

<b style="font-size:larger"><a href="/conekt/profile_comparison/">Compare Expression Profiles</a>:</b>
<p> Compare the expression of up to 50 genes on the same graph in each experimental condition or tissue.</p>

<b style="font-size:larger"><a href="/conekt/heatmap/">Gene Expression Heatmap</a>:</b>
<p> Visualize gene expression of a list of genes as a heatmap. The expression of a list of genes can be compared within a species using the default tab or between species using the comparative tab as a heatmap. The between species comparison can accept a list of more than 50 genes but is limited in the number of conditions or tissues visualized-- only for those tissue groups common to multiple species.</p>

<b style="font-size:larger"><a href="/conekt/search/specific/profiles">Find Genes Specifically Expressed in a Given Tissue</a>:</b>
<p>Using the form here genes can be found which are expressed specifically under a desired condition. Try SPM value 1.0 to list highly specific genes if exist.</p>

<b style="font-size:larger"><a href="/conekt/specificity_comparison/">Compare Specificity </a>:</b>
<p>Find out which genes are expressed specifically in species and/or conditions and have homologs expressed in a different species and/or condition.</p>

<b style="font-size:larger"><a href="/conekt/search/advanced">Expression of Gene Family Members</a>:</b>
<p>Use the Advanced Search tool to choose a species and input the exact gene family ID (select the available method; do not disable). Search will list all the available genes in the given gene family with links to each of their expression profiles.</p>
<hr/>
<section id="datasets">
<h3> Source of Data</h3>

Arachis:<br/>
<a href="https://doi.org/10.3389/fpls.2016.01446" target="_blank">Josh Clevenger, Ye Chu, Brian Scheffler, Peggy Ozias-Akins, A Developmental Transcriptome Map for Allotetraploid Arachis hypogaea, Front. Plant Sci. (2016)"</a>(<a href="https://www.ncbi.nlm.nih.gov/bioproject/PRJNA291488" target="_blank">PRJNA291488</a>)  
<br/>

Cicer: <br/> 
<a href="https://doi.org/10.1111/pbi.12059"  target="_blank"> Singh, V.K., Garg, R. and Jain, M. (2013), A global view of transcriptome dynamics during flower development in chickpea by deep sequencing. Plant Biotechnol J, 11: 691-701</a> (<a href="https://www.ncbi.nlm.nih.gov/bioproject/PRJNA182724" target="_blank">PRJNA182724</a>)
<br/>
Glycine: <br/>
<a href="https://doi.org/0.1111/j.1365-313X.2010.04222.x target="_blank"> Libault M et al., An integrated transcriptome atlas of the crop model Glycine max, and its use in comparative analyses in plants., Plant J, 2010 Apr 7;63(1):86-99</a>    
<br/>
Phaseolus: <br/>
<a href="https://doi.org/10.1186/1471-2164-15-866"  target="_blank">O'Rourke JA, Iniguez LP, Fu F, Bucciarelli B, Miller SS, Jackson SA, McClean PE, Li J, Dai X, Zhao PX, Hernandez G, Vance CP. An RNA-Seq based gene expression atlas of the common bean. BMC Genomics. 2014 Oct 6;15:866.</a> (<a href="https://www.ncbi.nlm.nih.gov/bioproject/PRJNA210619"  target="_blank">PRJNA210619</a>)
<br/>
Vigna: <br/>
<a href="https://doi.org/10.1111/tpj.13279" target="_blank">The Vigna unguiculata Gene Expression Atlas (VuGEA) from de novo assembly and quantification of RNA-seq data provides insights into seed maturation mechanisms</a> (<a href="https://www.ncbi.nlm.nih.gov/bioproject/PRJNA389300"  target="_blank">PRJNA389300</a>)
<br/>

<hr/>
</section>
<!--  ##
<b><a href="/conekt/"></a>:</b>
<b><a href="/conekt/"></a>:</b>
-->

<section id=about>

<!-- #########################################################  -->			

<h3/>About CoNekT</h3>
                    <p style="font-size: medium;">This is an instance of <a href="https://github.molgen.mpg.de/proost/CoNekT" target="_blank">CoNekT</a> (Coexpression Network Toolkit) at LIS, The <a href="https://legumeinfo.org" target="_blank">Legume Information System</a>.</p>
                    <p>It is work in progress at LIS and only some functionalities of CoNekT have been implimented here so far. </p>
                    <p><b>Publication for CoNekT:</b><br />CoNekT: an open-source framework for comparative genomic and transcriptomic network analyses (<a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6030989/" target="_blank">Proost and Mutwil, NAR 2018</a>).</p>
		    More information about CoNekT is available at <a href="https://github.molgen.mpg.de/proost/CoNekT" target="_blank">GitHub</a>(the original version) and more recently at <a href="https://github.com/sepro/CoNekT" target="_blank">https://github.com/sepro/CoNekT</a>.
		    <hr/>
                    <br /><br />
                    {% if g.msg %}
                    <div><span class="global-message"><strong>**{{ g.msg_title|safe }}: </strong>{{ g.msg|safe }}</span></div>
                    {% endif %}
                </div>
            </div>

        </div>

<!--  EXPERIMENTAL TOOL PANELS -->
<!--
	<div class="col-lg-3 col-md-3 col-sm-6 col-xs-6 panel-card" >
            <div class="panel panel-default">
	       <details>    
		       <summary><div class="panel-heading">Heatmap(of multiple genes) </div></summary>
		  
                <a href="{{ url_for('heatmap.heatmap_main') }}">
                    <img class="img-responsive img-card" src="{{ url_for('static', filename='img/cards/heatmap.png') }}" width="1150" height="1150"/>
                </a>
                <div class="panel-body">
                    <p>Allows users to <strong>compare the expression profiles</strong> from a custom set of genes in a heatmap</p>
                </div>
	      </details>
            </div>
        </div>
-->







        <!--<div class="col-lg-12">
            <div id="new-alert" class="alert alert-info alert-dismissible" role="alert" style="display: none">
                <button type="button" class="close" data-dismiss="alert" aria-label="Close"><span aria-hidden="true">&times;</span></button>
                <strong>New ?</strong> Check out our CoNekT's features <a href="{{ url_for('main.features') }}">here</a> ! </div>
            <div class="panel panel-default" style="background-color: #C1D0DE">
                <div class="panel-body">
                    <form  role="search" method="post" id="search-form-main" name="search-form-main" action="{{ url_for('search.simple') }}">
                        {{ g.search_form.csrf_token }}
                        <div>
                            <div class="from-group input-group">
                                {{ g.search_form.terms(class_="form-control", placeholder="Search") }}
                                <div class="input-group-btn"><button type="submit" class="btn btn-primary" value="Search" data-toggle="modal" data-target="#loaderModal">Search</button></div>
                            </div>
                            {% if keyword_examples %}
                            <span>e.g.
                                {% for ke in keyword_examples %}
                                <a href="{{ url_for('search.search_single_keyword', keyword=ke) }}">{{ ke }}</a>,
                                {% endfor %}
                                ...</span>
                            {% endif %}

                        </div>
                    </form>
                </div>
            </div>
        </div>-->
    </div>
    <!--<div class="row">
        <div class="col-lg-12">
          <div class="panel panel-default">
          <div id="featureCarousel" class="carousel slide" data-ride="carousel">
            <!-- Indicators ==>
            <ol class="carousel-indicators">
              <li data-target="#featureCarousel" data-slide-to="0" class="active"></li>
              <li data-target="#featureCarousel" data-slide-to="1"></li>
              <li data-target="#featureCarousel" data-slide-to="2"></li>
              <li data-target="#featureCarousel" data-slide-to="3"></li>
              <li data-target="#featureCarousel" data-slide-to="4"></li>
              <li data-target="#featureCarousel" data-slide-to="5"></li>
            </ol>

            <!-- Wrapper for slides ==>
            <div class="carousel-inner">
              <div class="item active">
                <div class="carousel-div">
                    <div class="row">
                        <div class="col-xs-2"><div style="margin-top:30px"><span style="color:#fff;font-size:36px;background-color:#337ab7;border-radius:4px;padding:10px"><i class="fa fa-search" ></i></span></div></div>
                        <div class="col-xs-10">
                            <h3>Quickly retrieve genes</h3>
                            <p>Use our <a href="{{ url_for('search.advanced') }}">Search</a> function {% if g.blast_enabled -%}or <a href="{{ url_for('blast.blast_main') }}">BLAST</a>{%- endif %} to quickly find genes of interest.</p>
                        </div>
                    </div>
                </div>
              </div>

              <div class="item">
                <div class="carousel-div">
                    <h3>Expression profiles and expression specificity</h3>
                    <p>See in which tissues and conditions a gene of interest is expressed or find genes <a href="{{ url_for('search.search_specific_profiles') }}">specifically expressed</a> in a certain tissue/condition.</p>
                </div>
              </div>

              <div class="item">
                <div class="carousel-div">
                    <h3>Co-expression networks</h3>
                    <p>Browse co-expression networks based on thousands of publicly available RNA-Seq samples! <a href="{{ url_for('search.search_enriched_clusters') }}">Find clusters</a> of co-expressed genes enriched for certain functions.</p>
                </div>
              </div>

              <div class="item">
                <div class="carousel-div">
                    <h3>Co-expression clusters</h3>
                    <p>Browse co-expression networks based on thousands of publicly available RNA-Seq samples! <a href="{{ url_for('search.search_enriched_clusters') }}">Find cluster</a> of co-expressed genes enriched for certain functions.</p>
                </div>
              </div>

              <div class="item">
                <div class="carousel-div">
                    <h3>Analyse custom gene sets</h3>
                    <p>Create <a href="{{ url_for('heatmap.heatmap_main') }}">heatmap</a> and <a href="{{ url_for('profile_comparison.profile_comparison_main') }}">expression profiles</a> for user defined sets of genes. Build a <a href="{{ url_for('custom_network.custom_network_main') }}">custom co-expression network</a> for a set of selected genes.</p>
                </div>
              </div>

              <div class="item">
                <div class="carousel-div">
                    <div class="col-xs-2"><div style="margin-top:30px"><span style="color:#fff;font-size:36px;background-color:#337ab7;border-radius:4px;padding:10px"><i class="fa fa-github"></i></span></div></div>
                    <div class="col-xs-10">
                        <h3>CoNekT is open-source</h3>
                        <p>Head to <a href="https://github.molgen.mpg.de/proost/CoNekT">GitHub</a> for a copy of the source if you want to host your own version with in-house data or other species!</p>
                    </div>
                </div>
              </div>
            </div>

            <!-- Left and right controls ==>
            <a class="left carousel-control" href="#featureCarousel" data-slide="prev">
              <span class="glyphicon glyphicon-chevron-left"></span>
              <span class="sr-only">Previous</span>
            </a>
            <a class="right carousel-control" href="#featureCarousel" data-slide="next">
              <span class="glyphicon glyphicon-chevron-right"></span>
              <span class="sr-only">Next</span>
            </a>
          </div>
        </div>
        </div>
    </div>-->
    <!--<div class="row">
        <div class="col-lg-6 col-md-6 col-sm-12">
            <div class="panel panel-success"  style="height:300px; overflow-y:scroll">
              <div class="panel-heading">News</div>
              <div class="panel-body">
                  {% for n in news %}
                  <div>
                      <p>{{ n.message_markup }}</p>
                      <p class="text-muted text-right">by {{ n.posted_by }} (<strong>{{n.posted_formatted}}</strong>)</p>
                      {% if not loop.last %}
                        <hr />
                      {% endif %}
                  </div>
                  {% else %}
                      <p><em>No news messages found</em></p>
                  {% endfor %}
              </div>
            </div>
        </div>
        {% if g.twitter_handle %}
        <div class="col-lg-6 col-md-6 col-sm-12">
            <div class="panel panel-default" style="height:300px; overflow-y:scroll">
                <a class="twitter-timeline" href="https://twitter.com/{{g.twitter_handle}}">Tweets by {{g.twitter_handle}}</a> <script async src="//platform.twitter.com/widgets.js" charset="utf-8"></script>
            </div>
        </div>
        {% endif %}
    </div>-->
</div>

</div>
{% endblock %}

{% block extrajs %}
<script src="{{ url_for('static', filename='js/js.cookie.js') }}"></script>
<script>
    $(document).ready(function() {
        // check cookie
        var visited = Cookies.get('visited');

        if (visited == null) {
            $('#new-alert').show();
        }

        // set cookie
        Cookies.set('visited', 'yes', { expires: 1000 });
    });
</script>
{% endblock %}
